File: src/main/java/com/tecsup/examen2/citas/service/CitaService.java
package com.tecsup.examen2.citas.service;

import org.springframework.stereotype.Service;
import com.tecsup.examen2.citas.repository.CitaRepository;
import com.tecsup.examen2.citas.model.Cita;
import com.tecsup.examen2.Pacientes.repository.PacienteRepository;
import com.tecsup.examen2.Medicos.repository.MedicoRepository;
import java.time.LocalDate;
import java.util.*;

@Service
public class CitaService {

    private final CitaRepository citaRepo;
    private final PacienteRepository pacienteRepo;
    private final MedicoRepository medicoRepo;

    public CitaService(CitaRepository citaRepo,
                       PacienteRepository pacienteRepo,
                       MedicoRepository medicoRepo) {
        this.citaRepo = citaRepo;
        this.pacienteRepo = pacienteRepo;
        this.medicoRepo = medicoRepo;
    }

    public List<Cita> findAll() {
        return citaRepo.findAll();
    }

    public Optional<Cita> findById(String id) {
        return citaRepo.findById(id);
    }

    public Cita create(Cita cita) {
        if (cita.getEstado() == null || cita.getEstado().isEmpty()) {
            cita.setEstado("PENDIENTE");
        }

        if (cita.getPaciente() != null && cita.getPaciente().getIdPaciente() != null) {
            pacienteRepo.findById(cita.getPaciente().getIdPaciente())
                    .orElseThrow(() -> new RuntimeException("Paciente no encontrado"));
        }

        if (cita.getMedico() != null && cita.getMedico().getIdMedico() != null) {
            medicoRepo.findById(cita.getMedico().getIdMedico())
                    .orElseThrow(() -> new RuntimeException("Médico no encontrado"));
        }

        return citaRepo.save(cita);
    }

    public Cita update(String id, Cita updated) {
        return citaRepo.findById(id).map(existing -> {
            existing.setFecha(updated.getFecha());
            existing.setHora(updated.getHora());
            existing.setMotivo(updated.getMotivo());
            existing.setEstado(updated.getEstado());
            return citaRepo.save(existing);
        }).orElseThrow(() -> new RuntimeException("Cita no encontrada"));
    }

    public void delete(String id) {
        citaRepo.deleteById(id);
    }

    public List<Cita> findByPaciente(String idPaciente) {
        return citaRepo.findByPacienteIdPaciente(idPaciente);
    }

    public List<Cita> findByMedico(String idMedico) {
        return citaRepo.findByMedicoIdMedico(idMedico);
    }

    public List<Cita> findByFecha(LocalDate fecha) {
        return citaRepo.findByFecha(fecha);
    }

    public List<Cita> findByEstado(String estado) {
        return citaRepo.findByEstado(estado);
    }

    public List<Cita> findCitasPendientesByPaciente(String idPaciente) {
        return citaRepo.findCitasPendientesByPaciente(idPaciente);
    }
}

File: frontend/src/services/citaService.js
import api from './api';

export const citaService = {
  getAll: async () => {
    const response = await api.get('/citas');
    return response.data;
  },

  getById: async (id) => {
    const response = await api.get(`/citas/${id}`);
    return response.data;
  },

  create: async (cita) => {
    const payload = {
      paciente: { idPaciente: cita.idPaciente },
      medico: { idMedico: cita.idMedico },
      fecha: cita.fecha,
      hora: cita.hora,
      motivo: cita.motivo,
      estado: cita.estado || 'PENDIENTE'
    };
    const response = await api.post('/citas', payload);
    return response.data;
  },

  update: async (id, cita) => {
    const response = await api.put(`/citas/${id}`, cita);
    return response.data;
  },

  delete: async (id) => {
    const response = await api.delete(`/citas/${id}`);
    return response.data;
  },

  getByPaciente: async (idPaciente) => {
    const response = await api.get(`/citas/paciente/${idPaciente}`);
    return response.data;
  },

  getByMedico: async (idMedico) => {
    const response = await api.get(`/citas/medico/${idMedico}`);
    return response.data;
  },

  getByFecha: async (fecha) => {
    const response = await api.get(`/citas/fecha/${fecha}`);
    return response.data;
  },

  getByEstado: async (estado) => {
    const response = await api.get(`/citas/estado/${estado}`);
    return response.data;
  },

  getPendientesByPaciente: async (idPaciente) => {
    const response = await api.get(`/citas/paciente/${idPaciente}/pendientes`);
    return response.data;
  }
};